JSON-Grammar-spec.md

```diff
@@ -0,0 +1,794 @@
+# JSON Grammar Specification
+
+A comprehensive specification for defining PEG (Parsing Expression Grammar) parsers using JSON syntax with the `jit-parser` library.
```

Comment:
Throughout this specification, do not mention `jit-parser`. The *JSON Grammar* is not designed just for `jit-parser` it is meant to be a generic specification.

Instead mention that it is a portable way of specifying grammars, which can be easily implemented in different libraries/languages.